packages/contractkit/src/wallets/ledger-wallet.ts

```diff
@@ -0,0 +1,222 @@
+
+export class LedgerWallet implements Wallet {
+  // Account addresses are hex-encoded, lower case alphabets
+  private readonly addressDerivationPath = new Map<Address, string>()
+  private addressesRetrieved = false
+  private setupFinished = false
+  private ledger: any
+
+  /**
+   * @param derivationPathIndexes number array of "address_index" for the base derivation path.
+   * Default: Array[0..9].
+   * Example: [3, 99, 53] will retrieve the derivation paths of
+   * [`${baseDerivationPath}/3`, `${baseDerivationPath}/99`, `${baseDerivationPath}/53`]
+   * @param baseDerivationPath base derivation path. Default: "44'/52752'/0'/0"
+   */
+  constructor(
+    readonly derivationPathIndexes: number[] = Array.from(Array(ADDRESS_QTY).keys()),
+    readonly baseDerivationPath: string = CELO_BASE_DERIVATION_PATH
+  ) {
+    const invalidDPs = derivationPathIndexes.some(
+      (value) => !(Number.isInteger(value) && value >= 0)
+    )
+    if (invalidDPs) {
+      throw new Error('ledger-wallet: Invalid address index')
+    }
+  }
+
+  /**
+   * @param transport Transport to connect the ledger device
+   */
+  async init(transport: any) {
+    try {
+      if (this.setupFinished) {
+        return
+      }
+      if (!this.ledger) {
+        this.ledger = new Ledger(transport)
+      }
+      if (!this.addressesRetrieved) {
+        debug('Fetching addresses from the ledger')
+        await this.retrieveAccounts()
+        this.addressesRetrieved = true
+      }
+      this.setupFinished = true
+    } catch (error) {
+      if (error instanceof TransportStatusError || error instanceof TransportError) {
+        this.transportErrorFriendlyMessage(error)
+      }
+      throw error
+    }
+  }
+
+  private async retrieveAccounts() {
+    const pairAddresses = await this.retrieveAccountsFromLedger()
+    pairAddresses.forEach((pair) =>
+      this.addressDerivationPath.set(normalizeAddressWith0x(pair.address), pair.derivationPath)
+    )
+  }
+
+  private async retrieveAccountsFromLedger() {
+    const addresses = []
+    // Each address must be retrieved synchronously, (ledger lock)
+    for (const value of this.derivationPathIndexes) {
+      const derivationPath = `${this.baseDerivationPath}/${value}`
+      const addressInfo = await this.ledger!.getAddress(derivationPath)
+      addresses.push({ address: addressInfo.address, derivationPath })
+    }
+    return addresses
+  }
+
+  getAccounts(): Address[] {
+    this.initializationRequired()
+    return Array.from(this.addressDerivationPath.keys())
+  }
+
+  hasAccount(address?: string) {
+    this.initializationRequired()
+    if (address) {
+      return this.addressDerivationPath.has(normalizeAddressWith0x(address))
+    } else {
+      return false
+    }
+  }
+
+  async signTransaction(txParams: Tx): Promise<EncodedTransaction> {
+    this.initializationRequired()
+    try {
+      const rlpEncoded = rlpEncodedTx(txParams)
+      const zrxInfo = await byContractAddress(txParams.to)
```

Comment:
It looks like this is pulling data from a static base64 encoded file to provide mapping from address to ERC20 metadata:
https://github.com/LedgerHQ/ledgerjs/blob/0471ef03e975d1430be2637193d37e8f529baddc/packages/hw-app-eth/src/erc20.js#L39
If this is true, this metadata would only be applicable for Ethereum mainnet. The ERC20s deployed to Celo network will have different addresses than Ethereum. In the future we could implement our own mappping, but I think we should remove this for now.

Reply:
🧐
@gastonponti

Reply:
Yes. I thought that I could add it for the future, and then if our ledger app supports contracts data, we could just make a fork of `hw-eth-app`, add the blob for the data of our contracts,  and just change the version in the `package.json`. But I removed it in other PR, just to make the code clearer. In the future if we need it, we will add it again. 👍

Reply:
PR: #3256